src/main/scala/com/gu/invoicing/refund/Impl.scala

```diff
@@ -213,12 +213,13 @@ object Impl {
 
   /** Select correct invoice to apply refund to */
   def decideRelevantInvoice(
+      refund: BigDecimal,
       invoices: List[Invoice],
       itemsByInvoiceId: Map[String, List[InvoiceItem]],
   ): (String, Invoice, List[InvoiceItem]) = {
     joinInvoiceWithInvoiceItemsOnInvoiceIdKey(invoices, itemsByInvoiceId).iterator
       .filter({ case (_, invoice, _) => invoice.Status == "Posted" })
-      .filter({ case (_, invoice, _) => invoice.Amount > 0 })
+      .filter({ case (_, invoice, _) => invoice.Amount >= refund })
```

Comment:
Maybe this should be `invoice.PaymentAmount` instead?

Reply:
good question, if they are in payment failure, (or the invoice is already adjusted or paid from a credit balance) then this might not work out well

Reply:
Actually existing adjustments and payments are fetched and checked later on in the program logic:
https://github.com/guardian/invoicing-api/blob/38fca38faa79761035975920e3a3ebdbd8687c4f/src/main/scala/com/gu/invoicing/refund/Program.scala#L44
https://github.com/guardian/invoicing-api/blob/38fca38faa79761035975920e3a3ebdbd8687c4f/src/main/scala/com/gu/invoicing/refund/Program.scala#L49
So it's probably ok, I wouldn't be surprised if there's a way to break it though!

Reply:
yeah but by that time we have already settled on a specific invoice which we fail if it doesn't have a payment

https://github.com/guardian/invoicing-api/blob/38fca38faa79761035975920e3a3ebdbd8687c4f/src/main/scala/com/gu/invoicing/refund/Program.scala#L49C27-L49C46
It doesn't look like we check if it's in payment failure, we would try to refund against a failed payment perhaps https://github.com/guardian/invoicing-api/blob/ccd5a34a5d0930efa6cd4fc0c81bd65012692256/src/main/scala/com/gu/invoicing/refund/Impl.scala#L49

the adjustment fetching is just so we can distribute further adjustments by the look